itest/rust/src/registration/constant_test.rs

```diff
@@ -0,0 +1,188 @@
+        .class_get_integer_constant_list_ex(class_name.clone().into())
+        .no_inheritance(true)
+        .done();
+
+    for (constant_name, constant_value) in CONSTANTS {
+        assert!(constants.contains(constant_name.into()));
+        assert_eq!(
+            ClassDb::singleton()
+                .class_get_integer_constant(class_name.clone().into(), constant_name.into()),
+            constant_value
+        );
+    }
+}
+
+#[derive(GodotClass)]
+struct HasOtherConstants {}
+
+impl HasOtherConstants {
+    const ENUM_NAME: &str = "SomeEnum";
+    const ENUM_A: i64 = 0;
+    const ENUM_B: i64 = 1;
+    const ENUM_C: i64 = 2;
+
+    const BITFIELD_NAME: &str = "SomeBitfield";
+    const BITFIELD_A: i64 = 1;
+    const BITFIELD_B: i64 = 2;
+    const BITFIELD_C: i64 = 4;
+}
+
+// TODO: replace with proc-macro api when constant enums and bitfields can be exported through the
+// proc-macro.
+impl ::godot::obj::cap::ImplementsGodotApi for HasOtherConstants {
+    fn __register_methods() {}
+    fn __register_constants() {
+        use ::godot::builtin::meta::registration::constant::*;
+        // Try exporting an enum.
+        ExportConstant::new(
+            ClassName::from_static("HasOtherConstants"),
+            ConstantKind::Enum {
+                name: Self::ENUM_NAME.into(),
+                variants: vec![
+                    IntegerConstant::new("ENUM_A".into(), Self::ENUM_A),
+                    IntegerConstant::new("ENUM_B".into(), Self::ENUM_B),
+                    IntegerConstant::new("ENUM_C".into(), Self::ENUM_C),
+                ],
+            },
+        )
+        .register();
+
+        // Try exporting an enum.
+        ExportConstant::new(
+            ClassName::from_static("HasOtherConstants"),
+            ConstantKind::Bitfield {
+                name: Self::BITFIELD_NAME.into(),
+                flags: vec![
+                    IntegerConstant::new("BITFIELD_A".into(), Self::BITFIELD_A),
+                    IntegerConstant::new("BITFIELD_B".into(), Self::BITFIELD_B),
+                    IntegerConstant::new("BITFIELD_C".into(), Self::BITFIELD_C),
+                ],
+            },
+        )
+        .register();
+    }
+}
+
+::godot::sys::plugin_add!(
+    __GODOT_PLUGIN_REGISTRY  in::godot::private;
+    ::godot::private::ClassPlugin {
+        class_name: "HasOtherConstants",
+        component: ::godot::private::PluginComponent::UserMethodBinds {
+            generated_register_fn: ::godot::private::ErasedRegisterFn {
+                raw: ::godot::private::callbacks::register_user_binds::<HasOtherConstants>,
+            },
+        },
+    }
+);
+
+macro_rules! test_enum_export {
+    (
+        // make the declaration look like other test-declarations, to make it easier to grep for and similar.
+        #[class = $class:ty]
+        #[enum_name = $enum_name:ident]
+        #[variants = [$($variant:ident),* $(,)?]]
+        #$attr:tt
+        fn $test_name:ident() { .. }
```

Comment:
Making things look like proc-macro attributes is rather confusing; I don't think we do that anywhere else (except one place working around a venial shortocming). I'd rather have more traditional key-value pairs like in the `packed_array.rs` macros...